src/util/finite_field.h

```diff
@@ -0,0 +1,195 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Alex Ozdemir
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * A finite-field element, implemented as a wrapper around Integer.
+ *
+ * TODOs:
+ * * consider montgomery form
+ * * extend to non-prime fields
+ */
+
+#include "cvc5_public.h"
+
+#ifndef CVC5__FINITE_FIELD_H
+#define CVC5__FINITE_FIELD_H
+
+#include <iosfwd>
+#include <iostream>
+
+#include "base/check.h"
+#include "base/exception.h"
+#include "util/integer.h"
+
+namespace cvc5::internal {
+
+class FiniteField
+{
+ public:
+  FiniteField(const Integer& val, const Integer& size)
+      // for the normalization
```

Comment:
What is this comment referring to?

Reply:
Ah. The floor computation. I've clarified to "normalize value into [0, size)".